packages/style-loader/index.js

```diff
@@ -1,10 +1,66 @@
-module.exports = ({ config }, options = {}) => config.module
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const merge = require('deepmerge');
+const { pick, mapValues } = require('lodash');
+
+module.exports = (neutrino, opts = {}) => {
+  const isBuild = neutrino.options.command === 'start';
+  const options = merge({
+    styleUseId: 'style',
+    cssHotLoaderUseId: 'cssHotLoader',
+    hot: true,
+    extract: {
+      plugin: {
+        filename: isBuild ? '[name].css' : '[name].[contenthash].css'
+      }
+    }
+  }, opts);
+
+  neutrino.config.module
   .rule(options.ruleId || 'style')
     .test(/\.css$/)
-      .use(options.styleUseId || 'style')
-        .loader(require.resolve('style-loader'))
-        .when(options.style, use => use.options(options.style))
-        .end()
-      .use(options.cssUseId || 'css')
-        .loader(require.resolve('css-loader'))
-        .when(options.css, use => use.options(options.css));
+    .use(options.styleUseId)
+       .loader(require.resolve('style-loader'))
+       .when(options.style, use => use.options(options.style))
+       .end()
+    .use(options.cssUseId || 'css')
+      .loader(require.resolve('css-loader'))
+      .when(options.css, use => use.options(options.css));
+
+  if (options.extract) {
+    const styleRule = neutrino.config.module.rule('style');
+    const styleEntries = styleRule.uses.entries();
+    const keys = Object.keys(styleEntries);
+    const useKeys = keys.filter(key => key !== options.styleUseId);
+    const fallbackKeys = keys.filter(key => key === options.styleUseId);
+    const loadersByKey = mapValues(styleEntries, use => ({
+      loader: use.get('loader'),
+      options: use.get('options')
+    }));
+
+    options.extract.loader = Object.assign({
+      use: Object.values(pick(loadersByKey, useKeys)),
+      fallback: Object.values(pick(loadersByKey, fallbackKeys))
+    }, options.extract.loader || {});
+
+    styleRule
+    .uses
+      .clear()
+      .end()
+    .when(options.hot, (rule) => {
+      rule.use(options.cssHotLoaderUseId)
+        .loader(require.resolve('css-hot-loader'))
+        .when(options.cssHotLoader, use => use.options(options.cssHotLoader));
+    });
+
+    ExtractTextPlugin.extract(options.extract.loader)
+    .forEach(({ loader, options }) => {
+      styleRule
+      .use(loader)
+        .loader(loader)
+        .options(options);
+    });
+
+    neutrino.config.plugin('extract')
```

Comment:
Move `.plugin` to the next line and increase indent 2 spaces from the line above it.